### PR TITLE
[DR-2796] Adding script for backfilling dates.

### PR DESCRIPTION
**Pull request by emu47**
## Jira Tickets ##
- [Backfill firstIndexed_dt and dateIndexed_dt](https://jira.nypl.org/browse/DR-2796)

## Things Done ##
- Added new one-off script for adding new field data to solr for indexed dates. 

## How to Test ##
- From app, run the script pointing to your solr of choice, e.g., `ruby ./scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb http://localhost:8983/solr/repoapi`
- It will run for a long time and may well need to be restarted. This is ok. It will get there eventually.

### CODE REVIEW

File: scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    total_results ||= response['response']['numFound']
+    
+    update_json = []
+
+    # Loop through each document
+    response['response']['docs'].each do |doc|
+      uuid = doc['uuid']
+
+      # Check if corresponding record exists in RepoSolrDoc
+      unless RepoSolrDoc.exists?(uuid: uuid)
```
Comment (croyfish) [inline scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb:45]: My local ingest DB doesn't have any of these.

 Is it possible that any records were pushed to solr straight from mms and wouldn't be in the production ingest database, or did we do the last full index via ingest?

Comment (croyfish) [reply scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb:45]: Is this script meant to run on the production server or from someone's local?

Comment (emu47) [reply scripts/one-offs/DR-2796-backfill-more-dates/backfill_additional_dates.rb:45]: This script is flexible, could be run from local, but I was thinking server. 

I can't say I expect any records to be missing from production database (and haven't found any yet), but I was being cautious?

**Review by croyfish — APPROVED** [review]
Let's give it a try. The logic makes sense.

**Pull request merged**